### PR TITLE
XmlDocument cleanup

### PR DESCRIPTION
**Pull request by Maxwe11**

### CODE REVIEW

File: src/Common/src/System/Xml/SecureStringHasher.cs [hunk 1]
```diff
@@ -11,7 +11,7 @@ namespace System.Xml
     // The SecureStringHasher implements IEqualityComparer for strings and therefore can be used in generic IDictionary.
     internal class SecureStringHasher : IEqualityComparer<String>
     {
-        int hashCodeRandomizer;
+        readonly int hashCodeRandomizer;
```
Comment (weshaggard) [inline src/Common/src/System/Xml/SecureStringHasher.cs]: This should also explicitly be marked as private.

**Comment by weshaggard** [comment]
@krwq please review these changes.

**Comment by Maxwe11** [comment]
re-committed in accordance with the review, so comments got deleted

**Comment by krwq** [comment]
Looks good to me :+1: Thank you for the change! @richamsft, could you also take a look?

File: src/Common/src/System/Xml/Base64Encoder.cs [hunk 2]
```diff
@@ -5,9 +5,9 @@ namespace System.Xml
 {
     internal abstract partial class Base64Encoder
     {
-        byte[] leftOverBytes;
-        int leftOverBytesCount;
-        char[] charsLine;
+        private byte[] leftOverBytes;
```
Comment (mellinoe) [inline src/Common/src/System/Xml/Base64Encoder.cs]: Adding explicit private modifiers is good, but is there a reason some of these fields didn't / don't follow the leading-underscore naming convention? I'd prefer to fix that all up at the same time unless there was a reason these didn't follow that convention in the first place.

Comment (mellinoe) [inline src/Common/src/System/Xml/Base64Encoder.cs]: This applies to all of the fields here, BTW. Some of the classes here do seem to follow the underscore convention so it's a little inconsistent.

File: src/System.Xml.XmlDocument/src/System/Xml/Dom/XmlLoader.cs [hunk 3]
```diff
@@ -523,12 +508,9 @@ private XmlAttribute LoadAttributeNodeDirect()
                 defattr.SetSpecified(false);
                 return defattr;
             }
-            else
-            {
-                attr = new XmlAttribute(r.Prefix, r.LocalName, r.NamespaceURI, _doc);
-                LoadAttributeValue(attr, true);
-                return attr;
-            }
+            attr = new XmlAttribute(r.Prefix, r.LocalName, r.NamespaceURI, _doc);
```
Comment (mellinoe) [inline src/System.Xml.XmlDocument/src/System/Xml/Dom/XmlLoader.cs]: I'm personally not a huge fan of removing these explicit else blocks, although it is functionally equivalent. I don't think we have any coding/style guidelines around this sort of thing.

File: src/Common/src/System/Xml/XmlCharType.cs [hunk 4]
```diff
@@ -111,7 +111,7 @@ internal struct XmlCharType
             "\u2070\u218f\u2c00\u2fef\u3001\ud7ff\uf900\ufdcf" +
             "\ufdf0\ufffd";
 #else
-        const string s_NCStartName =
+        private const string s_NCStartName =
```
Comment (mellinoe) [inline src/Common/src/System/Xml/XmlCharType.cs]: I think we usually just have Pascal-cased constants, but I can't find what our guidelines say about this right now... I see some other const's further down use Pascal-casing so at the very least we aren't being consistent here anyways. This is just an additional improvement that could be made, btw, the "private" modifier additions are good.

**Comment by mellinoe** [comment]
This looks good to me overall. I made a couple comments about places for further improvement (mostly field naming consistency) but those are just additional changes that could be made on top of these. :+1:

File: src/Common/src/System/Xml/Base64Encoder.cs (hunk 2 above)
Comment (krwq) [inline src/Common/src/System/Xml/Base64Encoder.cs]: Please do not make changes @mellinoe mentioned here. I will do it in separate commit. This is already tracked by: https://github.com/dotnet/corefx/issues/496

File: src/System.Xml.XmlDocument/src/System/Xml/Dom/XmlLoader.cs (hunk 3 above)
Comment (krwq) [inline src/System.Xml.XmlDocument/src/System/Xml/Dom/XmlLoader.cs]: I think the code is easier to read now. Please keep the changes

File: src/Common/src/System/Xml/XmlCharType.cs (hunk 4 above)
Comment (krwq) [inline src/Common/src/System/Xml/XmlCharType.cs]: Please leave renames for different review

File: src/System.Xml.XmlDocument/src/System/Xml/Dom/XmlLoader.cs (hunk 3 above)
Comment (eatdrinksleepcode) [inline src/System.Xml.XmlDocument/src/System/Xml/Dom/XmlLoader.cs]: I agree with @mellinoe: removing the else block makes it more difficult to understand the flow of the code just by looking at its structure. You have to spot the return statement at the end of the if block before you realize that the two blocks of code are mutually exclusive.

Can this be added to the code guidelines?

**Comment by weshaggard** [comment]
@Maxwe11 I would like to thank you for your interest and contributions and I think a lot of them are valuable. However there are also a lot of changes which I think should be reverted because they don't add enough value in comparison to the potential issues they might cause.

Similar to #482 with the else statements there are other type of changes in this PR that I would avoid making, such as all the changes to use the "?:" and "??" operators, the removal of nested ifs and default cases, as well as the conversion to auto-properties. As I stated in #482 at this point in time we still have an internal version of this source code as well and we need to keep changes in sync between them. So the amount of value those changes add is little to negative because the risk of merging source code is increased and the readability in some cases can be arguably worse. So we want to avoid making changes like these that don't have enough value to out weight the risk of introducing bugs. 

Also as stated in #482 we have an formatting tool that automatically does a lot of code clean-up, including things like explicitly marking fields as private and we have issue #496 which is to run it on the Xml code. 

For the above reasons I am going to revert this PR and I do want to apologize and thank you again for your interest and contributions.

I would still encourage you to submit a new PR that does some of the clean-up, like adding readonly to fields (note be careful of adding it to fields that are struct types, see [Mutating Readonly Structs](http://blogs.msdn.com/b/ericlippert/archive/2008/05/14/mutating-readonly-structs.aspx)) and eliminating some of the double casts.

**Pull request merged**